packages/fast-components-react-base/src/stack-panel/stack-panel.tsx

```diff
@@ -0,0 +1,738 @@
+    public componentWillUnmount(): void {
+        if (!isNil(this.rootElement.current)) {
+            if (((window as unknown) as WindowWithResizeObserver).ResizeObserver) {
+                this.resizeDetector.disconnect();
+                this.resizeDetector = null;
+            }
+        }
+    }
+
+    /**
+     * Create class-names
+     */
+    protected generateClassNames(): string {
+        const {
+            stackPanel,
+            stackPanel__isScrolling,
+        }: StackPanelClassNameContract = this.props.managedClasses;
+
+        return super.generateClassNames(
+            classNames(stackPanel, [stackPanel__isScrolling, this.state.isScrolling])
+        );
+    }
+
+    /**
+     * Render the items
+     */
+    private renderItems(): React.ReactFragment[] {
+        return React.Children.map(this.props.children, this.renderItem);
+    }
+
+    private isClonableElement(node: React.ReactNode): node is React.ReactElement<any> {
+        return React.isValidElement(node);
+    }
+
+    /**
+     * Render a single item
+     */
+    private renderItem = (item: React.ReactNode, index: number): React.ReactChild => {
+        if (
+            this.props.enableVirtualization &&
+            !this.props.neverVirtualizeIndexes.includes(index) &&
+            (index < this.state.renderedRangeStartIndex ||
+                index > this.state.renderedRangeEndIndex)
+        ) {
+            return;
+        }
+
+        const newStyleProps: React.CSSProperties = {
+            position: "absolute",
+            ...(this.props.orientation === stackPanelOrientation.horizontal
+                ? {
+                      width: toPx(this.itemPositions[index].span),
+                      left: toPx(this.itemPositions[index].start),
+                  }
+                : {
+                      height: toPx(this.itemPositions[index].span),
+                      top: toPx(this.itemPositions[index].start),
+                  }),
+        };
+
+        if (this.isClonableElement(item)) {
+            const styleProps: React.CSSProperties = get(item.props, "style");
+            return React.cloneElement(item, {
+                onFocusCapture: this.onItemFocus,
+                style: merge({}, styleProps, newStyleProps),
+            });
+        }
+
+        // it's a string, so wrap it in a div for positioning
+        return (
+            <div onFocusCapture={this.onItemFocus} style={newStyleProps}>
+                {item}
+            </div>
+        );
+    };
+
+    /**
+     * Builds a list of positioning data for all items
+     */
+    private updateItemPositions = (): void => {
+        const itemPositions: ItemPosition[] = [];
+        let currentItemTop: number = 0;
+        const childrenAsArray: React.ReactNode[] = React.Children.toArray(
+            this.props.children
+        );
+        const itemCount: number = childrenAsArray.length;
+        const keys: string[] = Object.keys(this.props.itemSpanOverrides);
```

Comment:
What's the benefit of the itemSpanOverrides not being just an array?

Reply:
It needs to be keyed as it only specifies the items that vary from the default span (ie. it does not need to include an entry for each item). I thought about using a Map but didn't think that seemed better.

Reply:
What I mean is, why do:
```
{
    "0": "foo",
    "1": "bar"
}
```

and not just:
```
[
    "foo",
    "bar"
]
```

Reply:
Because if we do a straight array we need to populate a value for every item, with keys we can limit to just those items that deviate from the default height.  Think a data grid that has a large number of rows but only the one the user is operating on is expanded.  With keys we can do:
 {
    "375": 200,
}

vs. 
[
    …, to the beginning
    100,
    100,
    200,
    100,
    100,
    … to the end
]

Reply:
You do not need to populate a real value for every item, if you did `var test = new Array(100)` you would get an array of 100 empty items.

Reply:
Replaced "defaultItemSpan" prop with simply "itemSpan" that takes either a single number for uniform sized items or a flat array (number[]) for variable sized items.  Seems more intuitive.

Reply:
Switched to a number[] array for simplicity.